src/components/GoonData/GoonData.tsx

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useEffect, useState } from "react";
+import { useAuthState } from "react-firebase-hooks/auth";
+import { auth, database } from "../../utils/firebase";
+import { onValue, ref } from "firebase/database";
+import { Button, Collapse, Grid, Paper, Typography, styled } from "@mui/material";
+import _ from "lodash";
+import { ExpandLess, ExpandMore } from "@mui/icons-material";
+
+const Item = styled(Paper)(({ theme }) => ({
+    backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
+    ...theme.typography.body2,
+    padding: theme.spacing(1),
+    textAlign: 'center',
+    color: theme.palette.text.secondary,
+}));
+
+const GoonData = () => {
+    const [user] = useAuthState(auth);
+    const [maps, setMaps] = useState<Record<string, number>>({});
+    const [latestVote, setLatestVote] = useState<string>("Unknown");
+    const [isExpanded, setIsExpanded] = useState<boolean>(false);
+
+    useEffect(() => {
+        if (!user) return;
+        const currentGoonRef = ref(database, "goons/current");
+        return onValue(currentGoonRef, (snapshot) => {
+            const firebaseData = snapshot.val();
+            if (firebaseData) {
+                setMaps(firebaseData["map"]);
+                setLatestVote(new Date(firebaseData["mostRecentVote"]).toLocaleString())
+            }
+        })
+    }, [user])
+
+    const toggleExpand = useCallback(() => {
+        setIsExpanded(!isExpanded);
+    }, [isExpanded])
+
+    const mostProbableMap = (_.maxBy(Object.entries(maps), "1") ?? ["Unknown"])[0];
```

Comment:
This is throwing an error on the preview build when I login

```
react_devtools_backend.js:2655 
        
       TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
```

Reply:
Error was database related.